mayastor/src/bdev/nexus/mod.rs

```diff
@@ -29,9 +35,79 @@ pub mod nexus_module;
 pub mod nexus_nbd;
 pub mod nexus_share;
 
+#[derive(Deserialize)]
+struct NexusShareArgs {
+    name: String,
+    protocol: String,
+    cntlid_min: u16,
+    cntlid_max: u16,
+}
+
+#[derive(Serialize)]
+struct NexusShareReply {
+    uri: String,
+}
+
 /// public function which simply calls register module
 pub fn register_module() {
-    nexus_module::register_module()
+    nexus_module::register_module();
+
+    jsonrpc_register(
+        "nexus_share",
```

Comment:
I'm wondering why don't we change grpc method instead of creating a whole new jsonrpc method.

Reply:
That was @gila's suggestion and that perhaps eventually gRPC methods will be replaced by JSON-RPC ones.

Reply:
until that happens I'm not sure it is reasonable to have the code duplicated. Having grpc nexus_share method and jsonrpc nexus_share method and both having different arguments is quite confusing already. Regardless if we decide to use jsonrpc or grpc in the future, if we start duplicating rpc methods now, it's going to make any change in the future more complicated. @gila any opinion?

Reply:
@jkryl I see your concern. But there is reason to the madness. What we want to do is figure out how we can, in a reasonable way -- implement a proper method without going out and breaking things all over the map by changing the bdev_share() and/or publish calls. So it is temporary really and subject to some churn as we go and figure out what's the best to do this.